xbout/geometries.py

```diff
@@ -180,19 +180,34 @@ def apply_geometry(ds, geometry_name, *, coordinates=None, grid=None):
         # Generates a coordinate whose value is 0 on the first grid point, not dz/2, to
         # match how BOUT++ generates fields from input file expressions.
         nz = updated_ds.dims[zcoord]
-        z0 = 2 * np.pi * updated_ds.metadata["ZMIN"]
-        z1 = z0 + nz * updated_ds.metadata["dz"]
-        if not np.isclose(
-            z1, 2.0 * np.pi * updated_ds.metadata["ZMAX"], rtol=1.0e-15, atol=0.0
-        ):
-            warn(
-                f"Size of toroidal domain as calculated from nz*dz ({str(z1 - z0)}"
-                f" is not the same as 2pi*(ZMAX - ZMIN) ("
-                f"{2.*np.pi*updated_ds.metadata['ZMAX'] - z0}): using value from dz"
+
+        # In BOUT++ v5, dz is either a Field2D or Field3D.
+        # We can use it as a 1D coordinate if it's a Field3D, _or_ if nz == 1
+        bout_v5 = updated_ds.metadata["BOUT_VERSION"] >= 5.0
+        use_metric_3d = updated_ds.metadata.get("use_metric_3d", False)
+        can_use_1d_z_coord = (nz == 1) or use_metric_3d
+
+        if can_use_1d_z_coord:
+            z = _1d_coord_from_spacing(updated_ds["dz"], zcoord)
```

Comment:
Will this work for a v4 data set with `nz == 1`? I think we need a way to get `updated_ds.metadata["dz"]` for that case.

Reply:
I think it should do, yes

Reply:
It did fail for v4 input with `nz==1` - I added a test which shows the error (b94e9bc), and then fixed it (456de1e).